File: Java/CollatzCalculator.java
class CollatzCalculator {

    int computeStepCount(int start) {
        if (start <= 0) {
            throw new IllegalArgumentException("Only positive integers are allowed");
        } else {
            int count_steps = 0;
            while (start != 1)
                {
                    if (start % 2 == 0) {
                        start = start / 2;
                    }
                    else {
                        start = (start * 3) + 1;
                    }
                    count_steps += 1;
                }
            return count_steps;
        }
    }
}

File: Kotlin/CollatzCalculator.kt
object CollatzCalculator {
    fun computeStepCount(start: Int): Int {
            var number: Int = start;
            if (number <= 0) {
            throw IllegalArgumentException("Only positive integers are allowed")
            return 0;
            } else {
            var count_steps: Int = 0;
            while (number != 1)
                {
                    if (number % 2 == 0) {
                        number = number / 2;
                        }
                    else {
                        number = (number * 3) + 1;
                        }
                    count_steps += 1;
                    }
            return count_steps;
    }
}
}